packages/dotnet/src/lib/core/dotnet.client.ts

```diff
@@ -24,160 +23,161 @@ import { LoadedCLI } from './dotnet.factory';
       parameters = swapKeysUsingMap(parameters, publishKeyMap);
-      const paramString = parameters ? getParameterString(parameters) : '';
-      cmd = `${cmd} ${paramString}`;
+      params.push(...getParameterArrayStrings(parameters));
     }
     if (publishProfile) {
-      cmd = `${cmd} -p:PublishProfile=${publishProfile}`;
+      params.push(`-p:PublishProfile=${publishProfile}`);
     }
     if (extraParameters) {
-      cmd = `${cmd} ${extraParameters}`;
+      params.push(`${extraParameters}`);
     }
-    return this.logAndExecute(cmd);
+    return this.logAndExecute(params);
   }
 
-  installTool(tool: string): void {
-    const cmd = `${this.cliCommand.command} tool install ${tool}`;
+  installTool(tool: string): Buffer {
+    const cmd = [`tool`, `install`, tool];
     return this.logAndExecute(cmd);
   }
 
-  restorePackages(project: string): void {
-    const cmd = `${this.cliCommand.command} restore "${project}"`;
+  restorePackages(project: string): Buffer {
+    const cmd = [`restore`, project];
     return this.logAndExecute(cmd);
   }
 
-  restoreTools(): void {
-    const cmd = `${this.cliCommand.command} tool restore`;
+  restoreTools(): Buffer {
+    const cmd = [`tool`, `restore`];
     return this.logAndExecute(cmd);
   }
 
-  format(project: string, parameters?: dotnetFormatOptions): void {
-    let cmd = `${this.cliCommand.command} format "${project}"`;
+  format(project: string, parameters?: dotnetFormatOptions): Buffer {
+    const params = [`format`, project];
     if (parameters) {
       parameters = swapKeysUsingMap(parameters, formatKeyMap);
-      const paramString = parameters ? getParameterString(parameters) : '';
-      cmd = `${cmd} ${paramString}`;
+      params.push(...getParameterArrayStrings(parameters));
     }
-    return this.logAndExecute(cmd);
+    return this.logAndExecute(params);
   }
 
   addProjectToSolution(solutionFile: string, project: string) {
-    const cmd = `${this.cliCommand.command} sln "${solutionFile}" add "${project}"`;
-    this.logAndExecute(cmd);
+    const params = [`sln`, solutionFile, `add`, project];
+    this.logAndExecute(params);
   }
 
   getSdkVersion(): Buffer {
-    const cmd = 'dotnet --version';
-    return this.execute(cmd);
+    return this.execute(['--version']);
   }
 
   printSdkVersion(): void {
-    this.logAndExecute('dotnet --version');
+    this.logAndExecute(['--version']);
   }
 
-  private logAndExecute(cmd: string): void {
-    console.log(`Executing Command: ${cmd}`);
-    execSync(cmd, { stdio: 'inherit', cwd: this.cwd || process.cwd() });
+  private logAndExecute(params: string[]): Buffer {
+    console.log(
+      `Executing Command: ${this.cliCommand.command} ${params.join(', ')}`,
+    );
+    return this.execute(params);
   }
 
-  private execute(cmd: string): Buffer {
-    return execSync(cmd, { cwd: this.cwd || process.cwd() });
+  private execute(params: string[]): Buffer {
+    return spawnSync(this.cliCommand.command, params, {
```

Comment:
Why are we swapping to spawnSync over execSync here? This is more complex maintainability-wise and shouldn't be needed.

Reply:
I'm not totally opposed to spawnSync actually, some stuff does work out better this way.

Reply:
Sorry, had a lengthy meeting couldn't look into this.

I converted to spawnSync to avoid converting all parameters to string first, spawnSync let us pass each parameter on the array so it is automatically escaped.
With execSync I had some problems passing on "Security Hotspot" tests because it allowed the user to pass arbitrary commands to execSync, like appending in the last parameter: "&& rm -rf *" this would delete everything on the current folder. Using the spawnSync command prevents the user to do this.

Reply:
That would be a lot easier because I had to change the method a lot, I'll take it into account next time, for this one I was kinda in a rush to have a deployable version because the previous one was broken.